### PR TITLE
Fix README inject example

### LINKED ISSUE #36: HOC inject example
Hi all.

I've been trying to find a way to create an `inject` helper function like in the README:

```
function inject(useSelector, baseComponent) {
    const store = useContext(StoreContext)
    const selected = useSelector(store)
    // optional memo essentially making a pure component
    return  baseComponent({ ...selected })
}
```

React throws a `Hooks can only be called inside the body of a function component.` error. I'm a bit of a noobie but haven't been able to find a way around this. It seems we may have to just use `useContext` and then a selector in the components themselves.

Can I please get your thoughts on this? Is my conclusion correct? Thanks :)

### PR DESCRIPTION
**Pull request by ross-weir**
fixes #36

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -292,27 +292,31 @@ function UiComponent({ name, age }) {
...
-export default () => {
-    // you may extract these two lines into a custom hook
-    const store = useContext(StoreContext)
-    const data = userSelector(store)
-    return UiComponent({...data})
-    // perhaps wrap it inside observer in here?
-    return observer(UiComponent({...data}))
-}
+export default () => (
+    () => {
```
Comment (FredyC) [inline README.md]: Well, this piece is correct actually :) It's just a regular component with hooks, it shouldn't be returning a function.

File: README.md [hunk 2]
```diff
@@ -292,27 +292,31 @@ function UiComponent({ name, age }) {
...
 // make universal HOC
 
-function inject(useSelector, baseComponent) {
-    const store = useContext(StoreContext)
-    const selected = useSelector(store)
-    // optional memo essentially making a pure component
-    return React.memo(props => baseComponent({ ...selected, ...props }))
-}
+const inject = (useSelector, baseComponent) => (
+    () => {
```
Comment (FredyC) [inline README.md]: I believe it should actually like this. With your approach, it's a component returning another component.

```suggestion
    React.useMemo((props) => {
        const store = useContext(StoreContext)
        const selected = useSelector(store)
        return baseComponent({ ...selected, ...props })
    })
```

Comment (ross-weir) [reply README.md]: Thank you. Hopefully it's looking OK now :+1:

Comment (FredyC) [reply README.md]: Copy & paste again, you forgot one thing. And try to think about what it actually does :)

Comment (ross-weir) [reply README.md]: Ops. Thanks for your patience. I'll be sure to do a lot more research!

File: README.md [hunk 3]
```diff
@@ -302,17 +302,19 @@ export default () => {
...
-    const store = useContext(StoreContext)
-    const selected = useSelector(store)
-    // optional memo essentially making a pure component
-    return React.memo(props => baseComponent({ ...selected, ...props }))
-}
+const inject = (useSelector, baseComponent) => (
+    React.useMemo((props) => {
+        const store = useContext(StoreContext)
+        const selected = useSelector(store)
+        // optional memo essentially making a pure component
```
Comment (FredyC) [inline README.md]: And this comment doesn't really fit here, does it? :)

**Pull request merged**